Update version of `buildGoModule` used in `flake.nix`

Whovian9369 opened the pull request:
Following on from #266, this should update the version of `buildGoModule` used to the version of `pkgs.buildGoModule` [that's in the repo's `flake.nix`.](https://github.com/packwiz/packwiz/blob/0c8beeac2fe7a771bd1ddae81d6737af470599de/flake.lock#L4-L11)

-----------

I'm also tempted to add some notes to the [`flake.nix`](https://github.com/packwiz/packwiz/blob/0c8beeac2fe7a771bd1ddae81d6737af470599de/flake.nix#L35), and update the notes in [`nix/prefetcher.nix`](https://github.com/packwiz/packwiz/blob/0c8beeac2fe7a771bd1ddae81d6737af470599de/nix/prefetcher.nix#L6-L13) to reflect this PR's change.

Specifically for [`nix/prefetcher.nix`](https://github.com/packwiz/packwiz/blob/0c8beeac2fe7a771bd1ddae81d6737af470599de/nix/prefetcher.nix#L7), I made a small error and mentioned the incorrect version of `pkgs.buildGoModule` that would be used... With the version of Nixpkgs that's currently pinned, [it should be `buildGo120Module`](https://github.com/NixOS/nixpkgs/blob/ff8d08cc60eb652ef261b180ce3707b2ff5579c5/pkgs/top-level/all-packages.nix#L25393-L25396) and not the `buildGo121Module` that I originally guessed.

I am also tempted to update the pinned version of Nixpkgs to [the latest commit that's currently available for `nixos-unstable`](https://github.com/NixOS/nixpkgs/commit/54aac082a4d9bb5bbc5c4e899603abfb76a3f6d6), but I'd want permission before doing that. (Possibly for exactly this PR?)

... Also it took me 45 minutes to write this, oops! 😅

comp500 commented:
> I'm also tempted to add some notes to the [`flake.nix`](https://github.com/packwiz/packwiz/blob/0c8beeac2fe7a771bd1ddae81d6737af470599de/flake.nix#L35), and update the notes in [`nix/prefetcher.nix`](https://github.com/packwiz/packwiz/blob/0c8beeac2fe7a771bd1ddae81d6737af470599de/nix/prefetcher.nix#L6-L13) to reflect this PR's change.

Go ahead, I don't use Nix myself so documenting what this stuff does is quite useful!

> I am also tempted to update the pinned version of Nixpkgs to [the latest commit that's currently available for `nixos-unstable`](https://github.com/NixOS/nixpkgs/commit/54aac082a4d9bb5bbc5c4e899603abfb76a3f6d6), but I'd want permission before doing that. (Possibly for exactly this PR?)

I think that would make sense, feel free to include that change in this PR.

Whovian9369 commented:
I added my notes and updated the Flake's pinned version for Nixpkgs...

One issue that I noticed is that, as I mentioned in #266...
> During testing I got a deprecation warning for `vendorSha256`, and that `vendorHash` should be used instead. 

I want to do some further digging into this, and I'll probably throw that into its own PR just so this one doesn't take longer than it absolutely needs to.

On top of that, I realized that `nix/prefetcher.nix` is *not* pinned against the same version that the `flake.nix` is (which is set in `flake.lock`), which is what caused #259 in the first place. I don't know more immediately how we "should" fix this, but it's something that I'm again going to look into for a future PR. (Perhaps with above mentioned `vendorHash` issue?)

I appreciate any thoughts that you may have!

Whovian9369 commented:
Decided to throw the `inputs.nixpkgs.url` edit in this PR so it's done, haha/

comp500 commented:
Thanks for the PR!
